src/workerd/api/worker-rpc.h

```diff
@@ -81,6 +81,115 @@ class JsRpcTarget: public jsg::Object {
   JSG_ONEWAY_SERIALIZABLE(rpc::SerializationTag::JS_RPC_STUB);
 };
 
+// Common superclass of JsRpcStub and Fetcher, the two types that may serve as the basis for
+// RPC calls.
+//
+// This class is NOT part of the JavaScript class heirarchy (it has no JSG_RESOURCE_TYPE block),
+// it's only a C++ class used to abstract how to get a capnp client out of the object.
+class JsRpcClientProvider: public jsg::Object {
+public:
+  // Get a capnp client that can be used to dispatch one call.
+  //
+  // If this isn't the root object (i.e. this is a JsRpcProperty), the property path starting from
+  // the root object will be appended to `path`.
+  virtual rpc::JsRpcTarget::Client getClientForOneCall(
+      jsg::Lock& js, kj::Vector<kj::StringPtr>& path) = 0;
```

Comment:
Non-blocking but I'm not a huge fan of using a `kj::Vector` for the path here. Something like a simple stack allocated struct would probably suffice?

```cpp
struct Path {
  kj::Maybe<Path&> parent = kj::none;
  kj::StringPtr name;
  kj::String toString();
}
```

So that this could be called simply as...

```
getClientForOneCall(js, Path { path, "foo"_kjc" })
```

But not that critical and don't feel too strongly about it.

Reply:
`path` is an "out" parameter. The callee fills it in, for the caller to use.

If we wanted to avoid any heap allocation here we could perhaps do it by tracking the "depth" of each property, so that we can pre-allocate the capnp list where the path is to be written, and then have `getClientForOneCall` directly fill that in. But, I don't think it's worth the complexity.

Reply:
ah, right, overlooked that part. ok.